arch/src/aarch64/gicv2.rs

```diff
@@ -0,0 +1,97 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{boxed::Box, result};
+
+use kvm_ioctls::{DeviceFd, VmFd};
+
+use super::gic::{finalize_device, init_device, set_device_attribute, Error, GICDevice};
+
+type Result<T> = result::Result<T, Error>;
+
+// Unfortunately bindgen omits defines that are based on other defines.
+// See arch/arm64/include/uapi/asm/kvm.h file from the linux kernel.
+const KVM_VGIC_V2_DIST_SIZE: u64 = 0x1000;
+const KVM_VGIC_V2_CPU_SIZE: u64 = 0x2000;
+
+/// Get the address of the GICv2 distributor.
+fn get_dist_addr() -> u64 {
```

Comment:
This can be declared in the `GICDevice` trait and defined separately for `GICv2` and `GICv3`, just like suggested for `init_device()`. Also please check if you can do something similar for all the other methods defined here. I would prefer to see them either declare in the `GICDevice` trait, or if they don't belong there you can define them directly in the `GICv2` struct. The same comment for all the methods defined in the `gicv3.rs` file outside of the `GICv3` struct.

Reply:
I sort of disagree on that. This is an implementation-specific attribute. It could be that another GIC implementation does not have a distributor address attribute, which would not allow it to implement the Trait

Reply:
Ok. I'm not very familiar with ARM and GIC. Feel free to move the methods where you think that it makes sense. If they are GIC generic, move them to the trait. If not move them inside GICv2 or GICv3 struct.